web/packages/hovercards/src/core.ts

```diff
@@ -120,6 +120,7 @@ export type Options = Partial< {
 	onFetchProfileFailure: OnFetchProfileFailure;
 	onHovercardShown: OnHovercardShown;
 	onHovercardHidden: OnHovercardHidden;
+	onCanShowHovercard: ( hash: string ) => boolean;
```

Comment:
We also provide event types for users:

1. Define a type for the event:

```tsx
export type OnCanShowHovercard = ( hash: string ) => boolean;

export type Options = Partial< {
	// other options...
	onCanShowHovercard: OnCanShowHovercard;
} >;
```

2. Export the type for both [core lib](https://github.com/Automattic/gravatar/blob/da7c8ac1f178012d86ae7fa72d20f94a2638313b/web/packages/hovercards/src/index.ts) and [react](https://github.com/Automattic/gravatar/blob/da7c8ac1f178012d86ae7fa72d20f94a2638313b/web/packages/hovercards/src/index.react.ts)

Reply:
Added the type and the option in the hook.